xdem/spstats.py

```diff
@@ -25,6 +27,109 @@
+
+    Workflow of the function:
+    Fills the no-data present on the regular N-D binning grid with nearest neighbour from scipy.griddata, then provides an
+    interpolant function that linearly interpolates/extrapolates using scipy.RegularGridInterpolator.
+
+    :param df: dataframe with statistic of binned values according to explanatory variables (preferably output of nd_binning)
+    :param list_var_names: explanatory variable data series to select from the dataframe (containing interval or float dtype)
+    :param statistic: statistic to interpolate, stored as a data series in the dataframe
+    :param min_count: minimum number of samples to be used as a valid statistic (replaced by nodata)
+    :return: N-dimensional interpolant function
+
+    :examples
+    Using a dataframe created from scratch
+    >>> df = pd.DataFrame({"var1": [1, 1, 1, 2, 2, 2, 3, 3, 3], "var2": [1, 2, 3, 1, 2, 3, 1, 2, 3], "statistic": [1, 2, 3, 4, 5, 6, 7, 8, 9]})
+        In 2 dimensions, the statistic array looks like this
+        array([
+            [1, 2, 3],
+            [4, 5, 6],
+            [7, 8, 9]
+            ])
+    >>> fun = interp_nd_binning(df, list_var_names=["var1", "var2"], statistic="statistic", min_count=None)
+    Right on point.
+    >>> fun((2, 2))
+        array(5.)
+    Interpolated linearly inside the 2D frame.
+    >>> fun((1.5, 1.5))
+        array(3.)
+    Extrapolated linearly outside the 2D frame.
+    >>> fun((-1, 1))
+        array(-5.)
+    """
+    # if list of variable input is simply a string
+    if isinstance(list_var_names,str):
+        list_var_names = [list_var_names]
+
+    # check that the dataframe contains what we need
+    for var in list_var_names:
+        if var not in df.columns:
+            raise ValueError('Variable "'+var+'" does not exist in the provided dataframe.')
+    statistic_name = statistic if isinstance(statistic,str) else statistic.__name__
+    if statistic_name not in df.columns:
+        raise ValueError('Statistic "' + statistic_name + '" does not exist in the provided dataframe.')
+    if min_count is not None and 'count' not in df.columns:
+        raise ValueError('Statistic "count" is not in the provided dataframe, necessary to use the min_count argument.')
+    if df.empty:
+        raise ValueError('Dataframe is empty.')
+
+    df_sub = df.copy()
+
+    # compute the middle values instead of bin interval if the variable is a pandas interval type
+    for var in list_var_names:
+        if isinstance(df_sub[var].dtype,pd.IntervalDtype):
+            df_sub[var] = pd.IntervalIndex(df_sub[var]).mid.values
+        # otherwise, leave as is
+
+    # keep only rows where the binning data exists for those variables (meaning they were used)
+    df_sub = df_sub[np.logical_and.reduce([np.isfinite(df_sub[var].values) for var in list_var_names])]
+    if df_sub.empty:
+        raise ValueError('Dataframe does not contain a nd binning with the variables corresponding to the list of variables.')
```

Comment:
Could this be more descriptive perhaps? This basically an implicit check for nans, so if this is empty, it means "dataframe has not a single valid row", right? Perhaps the `min_count is not None` logic could be placed above, and then the ValueError could say something along `f"Dataframe does not contain enough valid rows: {df_sub.shape[0]} {'threshold: ' + str(min_count) if min_count is not None else ''}."`

Reply:
The `min_count` replaces the `statistic` values as NaNs, not the binning values (e.g. slope of 5, 10, 15). This assertion simply verifies that there is binned values to work with.
I can add a check for the `statistic` after filtering with `min_count`

Reply:
Added in 8ed31e8